cpp/ql/src/experimental/Security/CWE/CWE-401/MemoryLeakOnFailedCallToRealloc.ql

```diff
@@ -13,6 +13,19 @@
 
 import cpp
 
+/**
+ * A function call that potentially does not return (such as `exit`).
+ */
+class CallMayNotReturn extends FunctionCall {
+  CallMayNotReturn() {
+    // call that is known to not return
+    not exists(this.(ControlFlowNode).getASuccessor())
+    or
+    // call to another function that may not return
+    exists(CallMayNotReturn exit | getTarget() = exit.getEnclosingFunction())
```

Comment:
I think this case is equivalent to:
```codeql
getTarget().calls(any(CallMayNotReturn exit))
```

And I find this easier to read. What do you think?

Reply:
Not quite, because `CallMayNotReturn` is a type of `FunctionCall` not `Function`.

Reply:
Oh, right. I guess it would be something like
```codeql
exists(CallMayNotReturn exit | getTarget().calls(exit.getTarget()))
```
and I guess at that point it's not any shorter.